etl/src/main/java/org/pih/hivmigration/etl/sql/HivExamOisMigrator.groovy

```diff
@@ -0,0 +1,248 @@
+            
+            -- Create Visit Diagnosis obs_group                                                                                           
+            INSERT INTO tmp_obs (
+                obs_id,
+                source_patient_id, 
+                source_encounter_id, 
+                concept_uuid)
+            SELECT 
+                x.obs_id,
+                x.source_patient_id,
+                x.source_encounter_id,
+                concept_uuid_from_mapping('PIH', 'Visit Diagnoses') as concept_uuid
+            from hivmigration_hiv_exam_ois x 
+            where x.oi is not null and x.oi != 'none'; 
+            
+            -- Mark it as PRESUMED diagnosis
+            INSERT INTO tmp_obs (
+                obs_group_id,
+                source_patient_id, 
+                source_encounter_id, 
+                concept_uuid,
+                value_coded_uuid)
+            SELECT 
+                x.obs_id,
+                x.source_patient_id,
+                x.source_encounter_id,
+                concept_uuid_from_mapping('PIH', 'CLINICAL IMPRESSION DIAGNOSIS CONFIRMED') as concept_uuid,
+                concept_uuid_from_mapping('PIH', 'PRESUMED') as value_coded_uuid
+            from hivmigration_hiv_exam_ois x 
+            where x.oi is not null and x.oi != 'none';  
+            
+            -- Set Diagnosis order to PRIMARY
+            INSERT INTO tmp_obs (
+                obs_group_id,
+                source_patient_id, 
+                source_encounter_id, 
+                concept_uuid,
+                value_coded_uuid)
+            SELECT 
+                x.obs_id,
+                x.source_patient_id,
+                x.source_encounter_id,
+                concept_uuid_from_mapping('PIH', 'Diagnosis order') as concept_uuid,
+                concept_uuid_from_mapping('PIH', 'primary') as value_coded_uuid
+            from hivmigration_hiv_exam_ois x 
+            where x.oi is not null and x.oi != 'none'; 
+            
+            -- Create Coded Diagnosis            
+            INSERT INTO tmp_obs (
+                obs_group_id,
+                source_patient_id, 
+                source_encounter_id, 
+                concept_uuid,
+                value_coded_uuid,
+                comments)
+            SELECT 
+                x.obs_id,
+                x.source_patient_id,
+                x.source_encounter_id,
+                concept_uuid_from_mapping('PIH', 'DIAGNOSIS') as concept_uuid,
+                concept_uuid_from_mapping(m.openmrs_concept_source, m.openmrs_concept_code) as value_coded_uuid,
+                x.comments
+            from hivmigration_hiv_exam_ois x, hivmigration_hiv_ois_mapping m  
+            where x.oi is not null and x.oi != 'none' and x.oi != 'other' and x.oi = m.oi 
+                and concept_uuid_from_mapping(m.openmrs_concept_source, m.openmrs_concept_code) is not null;  
```

Comment:
This seems like it could hide errors we'd want to be aware of. How about

```suggestion
                and m.openmrs_concept_code != '';  
```

Reply:
That's fine, but to be correct it should be:
m.openmrs_concept_source !='' and m.openmrs_concept_code != '';